cng/cng.go

```diff
@@ -68,19 +67,25 @@ func loadOrStoreAlg(id string, flags bcrypt.AlgorithmProviderFlags, mode string,
 }
 
 func utf16PtrFromString(s string) *uint16 {
-	str, err := syscall.UTF16PtrFromString(s)
-	if err != nil {
-		panic(err)
-	}
-	return str
+	return &utf16FromString(s)[0]
 }
 
+// utf16FromString converts the string using a stack-allocated slice of 32 bytes.
```

Comment:
Isn't it 64 bytes not 32? The element type is 2 bytes and it's a 32 length capacity.

Reply:
good catch, you are right!